pkgs/applications/misc/input-leap/default.nix

```diff
@@ -1,46 +1,57 @@
-{ lib, fetchFromGitHub, cmake, curl, xorg, avahi, qt5, openssl, stdenv
+{ lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, curl
+, xorg
+, avahi
+, qtbase
+, qttools
+, wrapQtAppsHook
+, openssl
+, stdenv
 , wrapGAppsHook
 , avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; }
-, fetchpatch }:
+, fetchpatch
+}:
 
 stdenv.mkDerivation rec {
   pname = "input-leap";
-  version = "2.4.0";
+  version = "unstable-2023-02-25";
 
   src = fetchFromGitHub {
-    owner = "input-leap";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-2tHqLF3zS3C4UnOVIZfpcuzaemC9++nC7lXgFnFSfKU=";
+    owner = "KoviRobi";
+    repo = "input-leap";
+    rev = "62ff806e08bdd2a06973175cbd36f4b08b9803ef";
+    sha256 = "lrNnvIbupK19ECy47Sffjb8zR4E35GxcQTdoae1Qnd0=";
     fetchSubmodules = true;
   };
 
-  patches = [
-    # This patch can be removed when a new version of inputleap (greater than 2.4.0)
-    # is released, which will contain this commit.
-    (fetchpatch {
-      name = "add-missing-cstddef-header.patch";
-      url =
-        "https://github.com/input-leap/inputleap/commit/4b12265ae5d324b942698a3177e1d8b1749414d7.patch";
-      sha256 = "sha256-ajMxP7szBFi4h8cMT3qswfa3k/QiJ1FGI3q9fkCFQQk=";
-    })
-  ];
-
   buildInputs = [
     curl
     xorg.libX11
     xorg.libXext
+    xorg.libXrandr
+    xorg.libXinerama
     xorg.libXtst
+    xorg.libXi
+    xorg.libICE
+    xorg.libSM
     avahiWithLibdnssdCompat
-    qt5.qtbase
+    qtbase
+    qttools
   ];
-  nativeBuildInputs = [ cmake wrapGAppsHook qt5.wrapQtAppsHook ];
+  nativeBuildInputs = [ cmake pkg-config wrapGAppsHook wrapQtAppsHook ];
 
-  postFixup = ''
-    substituteInPlace "$out/share/applications/inputleap.desktop" --replace "Exec=inputleap" "Exec=$out/bin/inputleap"
-  '';
```

Comment:
No need (e.g. if you do `nix build nixpkgs#xterm` and check in `result/share/applications/xterm.desktop` you'll see it has
```
Exec=xterm
Terminal=false
Type=Application
Encoding=UTF-8
Icon=xterm-color_48x48
```

But as it is on your branch, this fails for two reasons:
1. `inputleap.desktop` doesn't exist, because you are using the 2.4.0 branch
2. see https://github.com/input-leap/input-leap/pull/1609